### ISSUE TITLE
"double free" runtime error caused in cmor_axes.c 

### ISSUE THREAD

**Issue by IngoBethke** [PCMDI/cmor#216]
I got a runtime error (using intel/2017.1 compiler) caused by the line 
    free(cmor_axes[cmor_naxes].wrapping);
in cmor_axes.c

To fix it, I simply changed the code to
  if (refaxis.axis=='X') {
  free(cmor_axes[cmor_naxes].wrapping);
  }

I used cmor version 2.9.2 but it seems that the code in question is still the same in the latest version.

**Comment by taylor13** [comment]
Thanks for this bug report.  

If I may ask, are you planning to use CMOR for CMIP6?  If so, you will need to update to CMOR3.  Do you foresee any barriers to doing that?  How can we help?

**Comment by IngoBethke** [comment]
Thanks for offering help. We might get back to it soon. 

I'm testing CMOR3 right now for cmor-ization of output from the Norwegian Earth System Model.  So far everything looks good (using the Fortran interface) and I managed cmor-ize some test output. I envision that I will have fully upgraded our cmor-ization setup to CMOR3 and ready for CMIP6 production in a few weeks from now. Still, we might continue to use CMOR2 in parallel for a while for non-CMIP6 activities but will try to phase out its use.  

Some minor issues:

The cmor_axis procedure can cause an error if called with neither 'length' nor 'coord_vals' argument (e.g. when defining the time axis). If the length argument is not present then it is determined as size(coord_vals) without checking whether coord_vals is present. As a workaround, we simply specified length=1 when defining the time axis, which is strictly speaking not correct but does the job. 

With cmor2, I used to pre-load tables and then skip the table argument in successive cmor calls. I tried to do the same with cmor3  (e.g. CMIP6_grids.json) but somewhat surprisingly that trigged a lot of udunits and other errors. Not pre-loading the tables and instead specifying them in each cmor call solved the problem.    

Cheers, 
Ingo

**Comment by dnadeau4** [comment]
It seems that CMOR 3.0 has this code.  not sure if this help.  Also, it seems that `length` is always needed in `Fortran`.  

```
/* -------------------------------------------------------------------- */
/*     Fix issue #6 setting the pointer to NULL                         */
/* -------------------------------------------------------------------- */
    cmor_axes[cmor_naxes].wrapping = NULL;
```

**Comment by IngoBethke** [comment]
Thanks for clarifying and sorry for reporting a ‘bug’ that has already been corrected. I guess this issue can be closed then.
